File: MyMausam-2.0/packages/types/alerts.ts
export interface WeatherAlert {
  id: number;
  location_name: string;
  alert_type: string;
  severity: "green" | "yellow" | "orange" | "red" | string;
  description: string;
  date_of_issue: string;
  valid_upto: string;
  status_text: string;
  color?: string;
  is_active?: boolean;
}

export interface AlertFilter {
  severity?: string;
  location?: string;
  active_only?: boolean;
}

File: MyMausam-2.0/backend/python/app/schemas/alerts.py
from typing import Optional
from pydantic import BaseModel

class WeatherAlertSchema(BaseModel):
    id: Optional[int] = None
    location_name: str
    alert_type: str
    severity: str
    description: str
    date_of_issue: str
    valid_upto: str
    status_text: str
    color: Optional[str] = None
    is_active: bool = True
